src/main/java/com/hyperlocal/server/MerchantController.java

```diff
@@ -0,0 +1,73 @@
+package com.hyperlocal.server;
+
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+
+import com.github.jasync.sql.db.Connection;
+import com.github.jasync.sql.db.QueryResult;
+import com.github.jasync.sql.db.mysql.MySQLConnectionBuilder;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+// Rest Controller for Merchants
+
+@RestController
+public class MerchantController {
+
+  private static final String DATABASE_URL = "jdbc:mysql:///hyperlocal?socketFactory=com.google.cloud.sql.mysql.SocketFactory&cloudSqlInstance=speedy-anthem-217710:us-central1:hyperlocal";;
+  private static Connection connection;
+  private static final String MERCHANT_UPDATE_PREPARED_STATEMENT = "UPDATE `Merchants` SET `MerchantName` = ?, `MerchantPhone` = ? WHERE `MerchantID`= ?;";
+  private static final String MERCHANT_INSERT_PREPARED_STATEMENT = "INSERT into `Merchants` (`MerchantID`, `MerchantName`, `MerchantPhone`) values (?,?,?);";
+
+  public MerchantController() {
+    connection = MySQLConnectionBuilder.createConnectionPool(DATABASE_URL);
+  }
+
+  /*
+   * Update details of a merchant
+   * 
+   * Expects: A Json String from client as Input with the following format:
+   * 
+   * { "MerchantID": <Merchant sub id here>, "MerchantName": <Updated Name of the
+   * merchant here>, "MechantPhone": <Updated PhoneNumber of the merchant here> }
+   * 
+   * Returns: Merchant Object with updated Details if exists, else return inserted
+   * Merchant object
+   */
+
+  @PostMapping("/update/merchant/")
+  public CompletableFuture<String> updateMerchant(@RequestBody String postInputString) {
```

Comment:
Should these apis be returning strings ? I guess in that case, you should have to set the content type of the request to json somewhere

Reply:
Changed controller methods to return application/json